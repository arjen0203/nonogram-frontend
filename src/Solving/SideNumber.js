import React from "react";
import SolveNumberStyle from './NumberStyle.module.css';


class SideNumber extends React.Component {
    render() {
        return (
            <div className={SolveNumberStyle.ToBeSolvedSide}>{this.props.value}</div>
        );
    }
}

export default SideNumber;