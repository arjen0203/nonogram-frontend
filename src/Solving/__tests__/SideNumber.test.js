import React from "react";
import ReactDOM from 'react-dom';
import SideNumber from "../SideNumber";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const val = {number: 6, holds: false};

	const div = document.createElement('div');
	ReactDOM.render(<SideNumber value={val}></SideNumber>, div);
});

it('Render correctly', function () {
	const val = {number: 6, holds: false};

	const comp = render(<SideNumber value={val}></SideNumber>).getByTestId("sideNumber");
	expect(comp).toHaveTextContent("6");
	expect(comp).toHaveClass("to-be-solved-side");
});

it('Render correctly 2', function () {
	const val = {number: 3, holds: false};

	const comp = render(<SideNumber value={val}></SideNumber>).getByTestId("sideNumber");
	expect(comp).toHaveTextContent("3");
	expect(comp).toHaveClass("to-be-solved-side");
});

it('Render correctly is holding', function () {
	const val = {number: 6, holds: true};

	const comp = render(<SideNumber value={val}></SideNumber>).getByTestId("sideNumber");
	expect(comp).toHaveTextContent("6");
	expect(comp).toHaveClass("solved-side");
});

it('Render correctly is holding 2', function () {
	const val = {number: 3, holds: true};

	const comp = render(<SideNumber value={val}></SideNumber>).getByTestId("sideNumber");
	expect(comp).toHaveTextContent("3");
	expect(comp).toHaveClass("solved-side");
});

it('matches snapshot', function () {
	const val = {number: 6, holds: false};

	const tree = renderer.create(<SideNumber value={val}></SideNumber>).toJSON();
	expect(tree).toMatchSnapshot();
});