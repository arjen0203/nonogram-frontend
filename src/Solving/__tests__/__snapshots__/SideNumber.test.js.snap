// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="to-be-solved-side"
  data-testid="sideNumber"
>
  6
</div>
`;
